server/app.js

```diff
@@ -157,16 +160,17 @@ passport.use(
       ...PASSPORT_FIELDS_ATTRIBUTES,
       passReqToCallback: true,
     },
-    function (req, username, password, done) {
-      mongodb
-        .collection('users')
-        .findOne({ uid: username })
-        .then(function (err, user) {
-          if (!user) {
-            return done(null, false, req.body)
-          }
-          return done(null, true, null)
-        })
+    async function (req, username, password, done) {
+      try {
+        const { appDb } = req.app.locals
+        const { mail } = req.body
+        const userAttributes = { $or: [{ uid: username }, { mail }] }
```

Comment:
I think I'm understanding this change now. Is this the primary change (using `or` for username/email vs just looking up by username)?

Reply:
yes this is the primary change